src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -2231,10 +2231,14 @@ public static ExpressionSyntax GetStandaloneExpression(ExpressionSyntax expressi
 
             // In a well formed tree, the corresponding access node should be one of the ancestors
             // and its "?" token should precede the binding syntax.
-            while (currentNode != null)
+            while (true)
             {
                 currentNode = currentNode.Parent;
-                Debug.Assert(currentNode != null, "binding should be enclosed in a conditional access");
+
+                if (currentNode == null)
```

Comment:
This should be before the `currentNode.Parent` access.

Reply:
I don't understand. If you're talking about the line `currentNode = currentNode.Parent;`, then that both dereferences `currentNode` but also reassigns it and it's that reassigned value that needs to be checked for `null`. Or am I missing something?